markets/bfp-market/test/integration/modules/MarginModule.test.ts

```diff
@@ -2546,6 +2546,102 @@ describe('MarginModule', async () => {
     });
   });
 
+  describe('getWithdrawableMargin', () => {
+    it('should revert when accountId does not exist', async () => {
+      const { PerpMarketProxy } = systems();
+
+      const { marketId } = await depositMargin(bs, genTrader(bs));
+      const invalidAccountId = 42069;
+
+      await assertRevert(
+        PerpMarketProxy.getWithdrawableMargin(invalidAccountId, marketId),
+        `AccountNotFound("${invalidAccountId}")`,
+        PerpMarketProxy
+      );
+    });
+
+    it('should revert when marketId does not exist', async () => {
+      const { PerpMarketProxy } = systems();
+
+      const { trader } = await depositMargin(bs, genTrader(bs));
+      const invalidMarketId = 42069;
+
+      await assertRevert(
+        PerpMarketProxy.getWithdrawableMargin(trader.accountId, invalidMarketId),
+        `MarketNotFound("${invalidMarketId}")`,
+        PerpMarketProxy
+      );
+    });
+
+    it('should return zero when no collateral deposits', async () => {
+      const { PerpMarketProxy } = systems();
+
+      const { trader } = await genTrader(bs);
+      const { marketId } = genOneOf(markets());
+
+      const margin = await PerpMarketProxy.getWithdrawableMargin(trader.accountId, marketId());
+      assertBn.isZero(margin);
+    });
+
+    it('should return the full collateralUsd value when no position open', async () => {
+      const { PerpMarketProxy } = systems();
+
+      const desiredMarginUsdDepositAmount = genOneOf([5000, 10_000, 20_000]);
+      const { trader, marketId } = await depositMargin(
+        bs,
+        genTrader(bs, { desiredMarginUsdDepositAmount })
+      );
+
+      const margin = await PerpMarketProxy.getWithdrawableMargin(trader.accountId, marketId);
+      assertBn.near(margin, bn(desiredMarginUsdDepositAmount), bn(0.000001));
```

Comment:
This can probably be equal instead of near?

Reply:
Due to how collateral amounts are deposited, sometimes this can lead to values not totally exact so hence the `.near`

Reply:
Ah that's right, we're specifying usd amount